src/playermat/Playmat.ttslua

```diff
@@ -424,32 +409,54 @@ end
 
 function markSpawned(object)
   local saved = DATA_HELPER.call('setSpawnedPlayerCardGuid', {object.getGUID(), true})
-  if not saved then
-    error('attempt to mark player card spawned before data loaded')
-  end
+  if not saved then error('attempt to mark player card spawned before data loaded') end
 end
 
+-- contains position and amount of boxes for the upgradesheets that change uses
+-- Alchemical Distillation, Damning Testimony, Living Ink and Hyperphysical Shotcaster
+local customizationsTable = {
+  ["09040"] = {5, 2},
+  ["09059"] = {2, 2},
+  ["09079"] = {3, 2},
+  ["09119"] = {6, 4}
+}
+
 function spawnTokensFor(object)
   local cardMetadata = JSON.decode(object.getGMNotes()) or {}
-  local token = nil
   local type = nil
+  local token = nil
   local tokenCount = 0
-  if (cardMetadata.uses ~= nil) then
-    for i, useInfo in ipairs(cardMetadata.uses) do
-      token = useInfo.token
+  if cardMetadata.uses ~= nil then
+    for _, useInfo in ipairs(cardMetadata.uses) do
       type = useInfo.type
+      token = useInfo.token
       tokenCount = useInfo.count
-      if (activeInvestigatorId == "03004" and useInfo.type == "Charge") then
-        tokenCount = tokenCount + 1
+    end
+
+    -- additional uses for certain customizable cards (by checking the upgradesheets)
+    if customizationsTable[cardMetadata.id] ~= nil then
+      for _, obj in ipairs(searchArea(PLAY_ZONE_POSITION, PLAY_ZONE_SCALE)) do
+        local obj = obj.hit_object
+        if obj.tag == "Card" then
+          local notes = JSON.decode(obj.getGMNotes()) or {}
+          if notes.id == (cardMetadata.id .. "-c") then
+            local pos = customizationsTable[cardMetadata.id][1]
+            local boxes = customizationsTable[cardMetadata.id][2]
+            if obj.getVar("markedBoxes")[pos] == boxes then tokenCount = tokenCount + 2 end
+            break
+          end
+        end
       end
-      log("Spawning tokens for "..object.getName()..'['..object.getDescription()..']: '..tokenCount.."x "..token)
-      spawnTokenGroup(object, token, tokenCount)
     end
+
+    -- additional charge for Akachi
+    if activeInvestigatorId == "03004" and type == "Charge" then tokenCount = tokenCount + 1 end
+
+    log("Spawning tokens for "..object.getName()..'['..object.getDescription()..']: '..tokenCount.."x "..token)
+    spawnTokenGroup(object, token, tokenCount)
   else
     local data = getPlayerCardData(object)
-    if data == nil then
-      error('attempt to spawn tokens for ' .. object.getName() .. ': no token data')
-    end
+    if data == nil then error('attempt to spawn tokens for ' .. object.getName() .. ': no token data') end
```

Comment:
nil is expected if there's no tokens for the card, right?  Or am I misreading that?

Reply:
Yes, but then this function should not have been called (could get rid of this imo, but I left it there for now)